### PR TITLE
Bug fix: after clicking compare button, the dropdown button (for selecting different keys) is not shown

### PR DESCRIPTION
**Pull request by Lijiaoa**
### Description ###

- fix title bug
  - rename `Compare.tsx` to `CompareIndex.tsx`
  - delete compareIndex props: showDetails, because it could be got by its `title` property
  - delete compareIndex props: intermediateKeyList , deal with it in CompareIndex file

- about Trial class changes:
   - delete `final` and `finalAcc` field
   - delete `description` field (so change some other files)
   - add `parameter`(obj) field (so change some other files)
- about trialManager.ts 
   - add `allTrialsIntermediateChart` function (to filter Intermediate result all line chart data)
- Refactor Intermediate all line chart with the function: `allTrialsIntermediateChart `
- some little changes
  - hyper-parameter chart only render `succeed` trials
  - duration chart only render not `waiting` trials
     - update duration file `trialId` type: it's `number[]` rather than `string[]`
- about `TableList.tsx` file
  - get ret by `trial.tableRecord` 
- delete some interface: TableObj, Intermedia, Parameters, ErrorParameter
- delete interface:TableRecord's `accuracy` and `accDictionary` field because it's unuseful

#### Test Options ####
  - [x] test mnist-tfv2 experiment
  - [x] test final result is dict experiment
  - [x] test intermediate result is dict experiment
  - [ ] test nested search space experiment

### CODE REVIEW

File: ts/webui/src/components/experiment/trialdetail/chart/Intermediate.tsx [hunk 1]
```diff
@@ -89,20 +69,20 @@ class Intermediate extends React.Component<IntermediateProps, IntermediateState>
                     confine: true,
                     formatter: function (data: TooltipForIntermediate): React.ReactNode {
                         const trialId = data.seriesName;
-                        let parameters = {};
-                        let trialNum = 0;
-                        const temp = trialIntermediate.find(key => key.name === trialId);
-                        if (temp !== undefined) {
-                            parameters = temp.hyperPara;
-                            trialNum = temp.trialNum;
+                        let parameter = {}; // 这两个参数先行赋值因为有时候找不到对应的数据导致页面崩溃
```
Comment (ultmaster) [inline ts/webui/src/components/experiment/trialdetail/chart/Intermediate.tsx:72]: At least making this in English?

File: ts/webui/src/static/model/trialmanager.ts [hunk 2]
```diff
@@ -86,10 +87,37 @@ class TrialManager {
         return this.filter(trial => trial.status === 'SUCCEEDED');
     }
 
+    public unWaittingTrials(): Trial[] {
```
Comment (ultmaster) [inline ts/webui/src/static/model/trialmanager.ts:90]: notWaitingTrials

File: ts/webui/src/components/experiment/trialdetail/chart/Intermediate.tsx (hunk 1 above)
Comment (Lijiaoa) [reply ts/webui/src/components/experiment/trialdetail/chart/Intermediate.tsx:72]: yes!

File: ts/webui/src/static/model/trialmanager.ts (hunk 2 above)
Comment (Lijiaoa) [reply ts/webui/src/static/model/trialmanager.ts:90]: `notWaiting..` is better

**Review by ultmaster — APPROVED** [review]

File: ts/webui/src/static/interface.ts [hunk 3]
```diff
@@ -223,20 +186,25 @@ interface SearchItems {
     isChoice: boolean; // for parameters: type = choice and status also as choice type
 }
 
+interface allTrialsIntermediateChart {
+    name: string;
+    // id: string;
+    sequenceId: number;
+    data: number[]; // data名字是固定的
```
Comment (liuzhe-lz) [inline ts/webui/src/static/interface.ts:193]: No Chinese please. 🙂

**Review by liuzhe-lz — APPROVED** [review]

File: ts/webui/src/static/interface.ts (hunk 3 above)
Comment (Lijiaoa) [reply ts/webui/src/static/interface.ts:193]: yes!

**Pull request merged**